### PR TITLE
Replace _.isUndefined with native undefined check

### PR DESCRIPTION
**Pull request by vietduc01100001**
Addresses: #7747

### CODE REVIEW

File: lib/classes/Variables.js [hunk 1]
```diff
@@ -760,7 +760,7 @@ class Variables {
 
   buildOptions(regionSuffix) {
     const options = { useCache: true };
-    if (!_.isUndefined(regionSuffix)) {
+    if (regionSuffix !== undefined) {
```
Comment (medikoo) [inline lib/classes/Variables.js:763]: Let's replace it with `if (!regionSuffix)`, as in this context we rather check whether `regionSuffix` was passed, and we should discard any _falsy_ value

File: lib/classes/Variables.js [hunk 2]
```diff
@@ -984,7 +984,7 @@ class Variables {
       } else if (variableString.match(this.ssmRefSyntax)) {
         varType = 'SSM parameter';
       }
-      if (!_.isUndefined(varType)) {
+      if (varType !== undefined) {
```
Comment (medikoo) [inline lib/classes/Variables.js:987]: Same here

File: lib/plugins/aws/invokeLocal/index.js [hunk 3]
```diff
@@ -782,7 +782,7 @@ class AwsInvokeLocal {
       }
 
       const maybeThennable = lambda(event, context, callback);
-      if (!_.isUndefined(maybeThennable)) {
+      if (maybeThennable !== undefined) {
```
Comment (medikoo) [inline lib/plugins/aws/invokeLocal/index.js:785]: Same here

File: lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js [hunk 4]
```diff
@@ -114,7 +114,7 @@ module.exports = {
       );
     }
 
-    if (!_.isUndefined(apiGateway.minimumCompressionSize)) {
+    if (apiGateway.minimumCompressionSize !== undefined) {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js:117]: Same here

File: lib/plugins/aws/provider/awsProvider.js [hunk 5]
```diff
@@ -291,7 +291,7 @@ class AwsProvider {
 
     const request = this.requestQueue.add(() =>
       persistentRequest(() => {
-        if (options && !_.isUndefined(options.region)) {
+        if (options && options.region !== undefined) {
```
Comment (medikoo) [inline lib/plugins/aws/provider/awsProvider.js:294]: Same here

File: lib/utils/yamlAstParser.js [hunk 6]
```diff
@@ -90,7 +90,7 @@ const addNewArrayItem = (ymlFile, pathInYml, newValue) =>
     for (let i = 0; i < pathInYmlArray.length - 1; i++) {
       const propertyName = pathInYmlArray[i];
       const property = currentNode[propertyName];
-      if (_.isUndefined(property) || _.isObject(property)) {
+      if (property === undefined || _.isObject(property)) {
```
Comment (medikoo) [inline lib/utils/yamlAstParser.js:93]: Here we can remove _undefined_ check from condition (if value is an _object_ it certainly is not `undefined`)

File: lib/utils/yamlAstParser.js [hunk 7]
```diff
@@ -100,7 +100,7 @@ const addNewArrayItem = (ymlFile, pathInYml, newValue) =>
 
     const arrayPropertyName = _.last(pathInYmlArray);
     let arrayProperty = currentNode[arrayPropertyName];
-    if (_.isUndefined(arrayProperty) || Array.isArray(arrayProperty)) {
+    if (arrayProperty === undefined || Array.isArray(arrayProperty)) {
```
Comment (medikoo) [inline lib/utils/yamlAstParser.js:103]: Same here

File: lib/classes/Variables.js (hunk 1 above)
Comment (vietduc01100001) [reply lib/classes/Variables.js:763]: I'll carefully check this. Thanks!

File: lib/utils/yamlAstParser.js (hunk 6 above)
Comment (vietduc01100001) [reply lib/utils/yamlAstParser.js:93]: As I understand in this case, we allow `property` whose value is `undefined` or an *object* pass right? So we can't remove the undefined check here.

File: lib/utils/yamlAstParser.js (hunk 7 above)
Comment (vietduc01100001) [reply lib/utils/yamlAstParser.js:103]: Same as above.

File: lib/utils/yamlAstParser.js (hunk 6 above)
Comment (medikoo) [reply lib/utils/yamlAstParser.js:93]: Indeed, you're right. Sorry for confusion

File: lib/utils/yamlAstParser.js (hunk 7 above)
Comment (medikoo) [reply lib/utils/yamlAstParser.js:103]: Yes, that's correct. Ignore my comment

File: lib/utils/yamlAstParser.js [hunk 8]
```diff
@@ -90,7 +90,7 @@ const addNewArrayItem = (ymlFile, pathInYml, newValue) =>
     for (let i = 0; i < pathInYmlArray.length - 1; i++) {
       const propertyName = pathInYmlArray[i];
       const property = currentNode[propertyName];
-      if (_.isUndefined(property) || _.isObject(property)) {
+      if (!property || _.isObject(property)) {
```
Comment (medikoo) [inline lib/utils/yamlAstParser.js:93]: Here we now change behavior to not throw e.g. on `null` `false` and other falsy values.

Still I think it's acceptable, as as far I understand this code, the error fallback is purely to pick eventual programming error, which I believe is unlikely (we didn't see reports that it happens)

Comment (vietduc01100001) [reply lib/utils/yamlAstParser.js:93]: Unlikely means we're not 100% sure, right? I think it needs to be monitored to avoid unexpected behavior.

Comment (medikoo) [reply lib/utils/yamlAstParser.js:93]: While we introduced more permissive logic, there's no chance where scenario (for which it was changed) can occur. 

To me _else_ clause feels as dead code here (but haven't 100% confirmed on that)

**Pull request merged**